Feature/DGI-41: Generate and validate SSO Token

ntw-ITM opened the pull request:
I'm not sure how long the tokens should be valid for. For now, I've set it to 7 days, but I guess that may be too long.

objmj-itminds commented on backend/Web/Services/TokenService.cs:
```diff
@@ -36,5 +41,66 @@ public string CreateToken(IUser user)
+        throw new NotFoundException(nameof(User), user.Id);
+      }
+
+      userEntity.SSOTokenId = token.Id;
+      _context.SaveChanges();
```
Instead of having this service mutating and writing to db instead, return a tuple of the id and written token.

objmj-itminds commented on backend/Web/Services/TokenService.cs:
```diff
@@ -36,5 +41,66 @@ public string CreateToken(IUser user)
+        ValidateAudience = false
+      }, out SecurityToken validatedToken);
+
+      var user = await _context.Users.FindAsync(((JwtSecurityToken)validatedToken).Claims.First(claim => claim.Type == ClaimTypes.NameIdentifier).Value);
```
We don't want any DB reads in this service. Simply return the user identifier.

objmj-itminds commented on frontend/src/services/backend/nswagts.ts:
```diff
@@ -1110,6 +1110,7 @@ export class User extends AuditableEntity implements IUser {
     deactivationTime?: Date | null;
     accountId?: number;
     account?: Account | null;
+    ssoTokenId?: string | null;
```
I don't like that this ID is here, and I can't find it in any committed DTO. I don't think the frontend needs this ID.

ntw-ITM commented:
Added requested changes to TokenService. Also Added Newtonsoft.Json to Domain project to use JsonIgnore on ssoTokenId property.

objmj-itminds commented on backend/Domain/Entities/AdminUser.cs:
```diff
@@ -12,5 +13,7 @@ public class AdminUser : AuditableEntity, IUser
     public RoleEnum Role { get; set; }
     public string Name { get; set; }
     public DateTimeOffset? DeactivationTime { get; set; }
+    [JsonIgnore]
```
I didn't mean to pull logic into your domain. 

The issue is you are using IUser outside the domain which can cause a reference to the attribute.

objmj-itminds reviewed (changes requested):
Mistakenly approved.

ntw-ITM commented:
I have now removed the JsonIgnores again and uninstalled Newtonsoft.Json from Domain project. I have removed the SSOTokenId property from IUser, but the issue with it being generated in nswagts.ts has not been resolved.
